### PR TITLE
add admin apis for bulk set severity

### PR DESCRIPTION
**Pull request by refaelm92**
## PR Type:
Enhancement

___
## PR Description:
This PR introduces two new APIs to allow bulk updating of severity scores for vulnerability and posture exceptions. The main changes include:
- Addition of two new functions in `db/utils.go` to handle the bulk update operations: `BulkSetVulnerabilityExceptionsSeverity` and `BulkSetPostureExceptionsSeverity`.
- Addition of two new routes in `routes/v1/admin/routes.go` to handle the PUT requests for the bulk update operations: `updateVulnerabilityExceptionsSeverity` and `updatePostureExceptionsSeverity`.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

- `db/utils.go`: Added two new functions `BulkSetVulnerabilityExceptionsSeverity` and `BulkSetPostureExceptionsSeverity` to perform bulk update operations on the severity scores of vulnerability and posture exceptions respectively.
- `routes/v1/admin/routes.go`: Added two new routes `updateVulnerabilityExceptionsSeverity` and `updatePostureExceptionsSeverity` to handle the PUT requests for the bulk update operations.
</details>

___
## User Description:
Signed-off-by: refaelm <refaelm@armosec.io>

### CODE REVIEW

File: db/utils.go [hunk 1]
```diff
@@ -27,6 +27,30 @@ func ValidateCollection(collection string) error {
 //////////////////////////////////Sugar functions for mongo using values in gin context /////////////////////////////////////////
 /////////////////////////////////all methods are expecting collection and customerGUID from context/////////////////////////////
 
+func BulkSetVulnerabilityExceptionsSeverity(c context.Context, cves []string, severity int) error {
```
Comment (avrahams) [inline db/utils.go:30]: In the DB level we should have generic functions 
So here you can define a function ```AdminUpdateMany``` that gets filter and update command 
the collection should pass as value of the context as we do the the rest of the places
all the vulnerability / posture specific query should be in the admin handler

**Review by avrahams — CHANGES_REQUESTED** [review]

File: routes/v1/admin/routes.go [hunk 2]
```diff
@@ -43,12 +43,49 @@ func AddRoutes(g *gin.Engine) {
...
+	admin.PUT("/updatePostureExceptionsSeverity", updatePostureExceptionsSeverity)
+
 	//Post V2 list query on other collections
 	admin.POST("/:path/query", adminSearchCollection)
 	//uniqueValues
 	admin.POST("/:path/uniqueValues", adminAggregateCollection)
 }
 
+func updateVulnerabilityExceptionsSeverity(c *gin.Context) {
+	updateReq := struct {
```
Comment (avrahams) [inline routes/v1/admin/routes.go:56]: Lets have this in types and not as a private type

File: db/utils.go [hunk 3]
```diff
@@ -27,6 +27,20 @@ func ValidateCollection(collection string) error {
...
+func AdminUpdateMany(c context.Context, filter *FilterBuilder, update bson.D) error {
+	defer log.LogNTraceEnterExit("AdminUpdateMany", c)()
+	collection, err := readCollection(c)
+	if err != nil {
+		return err
+	}
+
+	res, err := mongo.GetWriteCollection(collection).UpdateMany(c, filter.get(), update)
+	if res != nil {
+		log.Log(fmt.Sprintf("AdminUpdateMany: updated %d documents", res.ModifiedCount), c)
```
Comment (avrahams) [inline db/utils.go:39]: Don't we want to return the modified count and also return it in the response

File: routes/v1/admin/routes.go [hunk 4]
```diff
@@ -43,12 +43,52 @@ func AddRoutes(g *gin.Engine) {
 	//add delete customers data route
 	admin.DELETE("/customers", deleteAllCustomerData)
 
+	admin.PUT("/updateVulnerabilityExceptionsSeverity",
```
Comment (avrahams) [inline routes/v1/admin/routes.go:46]: Can we add tests - there is a special set for admin tests

**Review by avrahams — CHANGES_REQUESTED** [review]

File: db/utils.go (hunk 3 above)
Comment (refaelm92) [reply db/utils.go:39]: it's going to be a cron job, besides logging it we don't have much to do with it

**Review by avrahams — APPROVED** [review]

**Pull request merged**